scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -145,13 +145,47 @@ class FossId(
             .build()
 
         service = retrofit.create(FossIdRestService::class.java)
+
+        version = runBlocking {
+            parseVersion().orEmpty()
+        }
     }
 
     override fun filterOptionsForResult(options: ScannerOptions) =
         options.mapValues { (k, v) ->
             v.takeUnless { k in secretKeys }.orEmpty()
         }
 
+    /**
+     * Extract the version version from the login page.
+     * Example: &nbsp;&nbsp;&nbsp;cli.  3.1.16 (build 5634934d, RELEASE)
+     */
+    private suspend fun parseVersion(): String? {
+        // TODO: replace with an API call when FossID provides a function (starting at version 21.2).
+        val regex = Regex("^.*&nbsp;(cli. *[0-9\\. ]+\\(build[\\w, ]+\\)).*$")
+
+        val response = service.getLoginPage()
+
+        response.use {
+            it.charStream().buffered().apply {
+                this.useLines { lines ->
+                    lines.forEach { line ->
+                        val matcher = regex.matchEntire(line)
+                        if (matcher != null) {
+                            if (matcher.groupValues.size == 2) {
```

Comment:
Use `matcher != null && matcher.groupValues.size == 2` to reduce nesting.

Reply:
Or even shorter: `matcher?.groupValues.size == 2`